### PR TITLE
proxy: Put the pb object into the struct

### PR DESCRIPTION
**Pull request by SimFG**
Avoid repeated creation of pb objects.

Signed-off-by: SimFG <1142838399@qq.com>

### CODE REVIEW

File: server/proxy/grpcproxy/auth.go [hunk 1]
```diff
@@ -18,98 +18,81 @@ import (
 	"context"
 
 	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
-	"go.etcd.io/etcd/client/v3"
+	clientv3 "go.etcd.io/etcd/client/v3"
 )
 
 type AuthProxy struct {
-	client *clientv3.Client
+	auth pb.AuthClient
```
Comment (ahrtr) [inline server/proxy/grpcproxy/auth.go:25]: suggest to rename to `authClient`

File: server/proxy/grpcproxy/election.go [hunk 2]
```diff
@@ -17,35 +17,34 @@ package grpcproxy
...
 	"context"
 
-	"go.etcd.io/etcd/client/v3"
+	clientv3 "go.etcd.io/etcd/client/v3"
 	"go.etcd.io/etcd/server/v3/etcdserver/api/v3election/v3electionpb"
 )
 
 type electionProxy struct {
-	client *clientv3.Client
+	election v3electionpb.ElectionClient
```
Comment (ahrtr) [inline server/proxy/grpcproxy/election.go:25]: suggest to rename to `electionClient`

File: server/proxy/grpcproxy/lock.go [hunk 3]
```diff
@@ -17,22 +17,22 @@ package grpcproxy
...
 	"context"
 
-	"go.etcd.io/etcd/client/v3"
+	clientv3 "go.etcd.io/etcd/client/v3"
 	"go.etcd.io/etcd/server/v3/etcdserver/api/v3lock/v3lockpb"
 )
 
 type lockProxy struct {
-	client *clientv3.Client
+	lock v3lockpb.LockClient
```
Comment (ahrtr) [inline server/proxy/grpcproxy/lock.go:25]: suggest to rename to `lockClient`

File: server/proxy/grpcproxy/maintenance.go [hunk 4]
```diff
@@ -19,32 +19,30 @@ import (
 	"io"
 
 	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
-	"go.etcd.io/etcd/client/v3"
+	clientv3 "go.etcd.io/etcd/client/v3"
 )
 
 type maintenanceProxy struct {
-	client *clientv3.Client
+	maintenance pb.MaintenanceClient
```
Comment (ahrtr) [inline server/proxy/grpcproxy/maintenance.go:26]: `maintenanceClient`

File: server/proxy/grpcproxy/auth.go (hunk 1 above)
Comment (SimFG) [reply server/proxy/grpcproxy/auth.go]: Done. Thanks for your suggestions. I have tow questions in the discussions. If you are free, I hope you can help me.

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you!

**Review by serathius — APPROVED** [review]

**Pull request merged**